client/src/VideoCards.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import "./VideoCards.css";
+
+const VideoCards = ({ videos, setVideos }) => {
+    const handleDeleteClick = (id) => {
+        fetch(`http://localhost:5000/${id}`,{
+            method: "DELETE",
+            headers: {
+                "Content-Type": "application/json",
+            },
+        })
+        .then((response) => response.json())
+        .then((data) =>{
+            console.log("deleted", data)
+            let newData = videos.filter((video) => video.id !== id);
+            setVideos(newData);
+        })
+        .catch((error) => {
+            console.error("Error:", error);
+        });
+};
+
+const increaseRating = (id) => {
+    let newData = [];
+    newData = videos.map((video) => 
+    video.id === id ? { ...video, rating: video.rating + 1 }: video);
+    fetch(`http://localhost:5000/rating`,{
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({ id, increment: 1 }),
+        })
+        .catch((error) => {
+            console.error("Error:", error);
+        });
+    return setVideos(newData);
+}
+
+const decreaseRating = (id) => {
```

Comment:
The increaseRating and decreaseRating functions work fine; as there's so little difference between them you could use a shared function and pass it a parameter that indicates whether to increment or decrement.

Reply:
yes, i have done it. thank you for you suggestions.